sql/partition_info.cc

```diff
@@ -871,26 +872,13 @@ bool partition_info::vers_init_info(THD * thd)
   return false;
 }
 
-bool partition_info::vers_set_interval(const INTERVAL & i)
+bool partition_info::vers_set_interval(int interval, interval_type type)
 {
-  if (i.neg || i.second_part)
-    return true;
-
   DBUG_ASSERT(vers_info);
-
-  // TODO: INTERVAL conversion to seconds leads to mismatch with calendar intervals (MONTH and YEAR)
-  vers_info->interval= static_cast<my_time_t>(
-    i.second +
-    i.minute * 60 +
-    i.hour * 60 * 60 +
-    i.day * 24 * 60 * 60 +
-    i.month * 30 * 24 * 60 * 60 +
-    i.year * 365 * 30 * 24 * 60 * 60);
-
-  if (vers_info->interval == 0)
+  if (type > INTERVAL_SECOND)
     return true;
-
-  return false;
+  vers_info->interval= Interval(interval, type);
+  return !vers_info->interval;
```

Comment:
This is checked in parser. Actually, this method is not needed much since all checks are done now outside of it.